### DISCUSSION TITLE
Difference in Directory Listings between PowerShell and Python

### DISCUSSION THREAD

**Discussion by gyuta** [python-poetry/poetry#9764] [Q&A]
#### Description:
I recently installed Poetry on Windows and everything seemed to install correctly. However, while trying to set the PATH, I couldn't find the relevant directory. After some trial and error, I noticed a discrepancy in the directory listings between PowerShell and Python.

```powershell
PS C:\Users\me> (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | python -
Retrieving Poetry metadata

The latest version (1.8.3) is already installed.
```

When listing the contents of the `C:\Users\me\AppData\Roaming` directory, there are discrepancies between the outputs of PowerShell and Python. 

#### PowerShell Output:
```powershell
PS C:\Users\me> Get-ChildItem C:\Users\me\AppData\Roaming\

    Directory: C:\Users\me\AppData\Roaming

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----        2024/06/14      9:53                Adobe
d-----        2024/06/14      9:54                Ahkun.Pasolog.Google.Apis.Auth
d-----        2024/10/13     11:48                Code
d-----        2024/07/16     11:56                DBeaverData
d---s-        2024/10/02     14:45                Microsoft
d-----        2024/07/09     11:21                teraterm5
```

#### Python Output:
```python
import os
print(os.listdir("C:/Users/me/AppData/Roaming/"))
# Output: ['Adobe', 'Ahkun.Pasolog.Google.Apis.Auth', 'Code', 'DBeaverData', 'Microsoft', 'pypoetry', 'Python', 'teraterm5']
```

#### Observations:
1. **Differences in Output**:
   - The Python output includes additional directories: `pypoetry` and `Python`.
   - PowerShell's output shows the `Microsoft` directory with special attributes (`d---s-`).

2. **Potential Issues**:
   - The difference in outputs might be due to:
     - Hidden or system files not being displayed in PowerShell by default.
     - Access permissions affecting what is listed.
     - Different execution contexts (user accounts, privileges) for PowerShell and Python.

#### Request for Help:
- Can anyone provide insights into why there might be discrepancies between the two outputs? 
- Are there specific configurations or commands in PowerShell that would allow for consistent listings with Python's `os.listdir()`?

**Comment by schneebuzz** [comment]
> Are there specific configurations or commands in PowerShell that would allow for consistent listings with Python's os.listdir()?

By default, `Get-ChildItem` does not include empty or hidden directories/files. See [Get-ChildItem](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-childitem?view=powershell-7.4). There you can also see that the `s` in the listing is for `system`.

**Reply by gyuta** [reply]
Thank you for your response! However, even after showing hidden folders, the situation remains unchanged...

```
PS C:\Users\me> Get-ChildItem C:\Users\me\AppData\Roaming\ -force


    Directory: C:\me\20057\AppData\Roaming


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----        2024/06/14      9:53                Adobe
d-----        2024/06/14      9:54                Ahkun.Pasolog.Google.Apis.Auth
d-----        2024/10/15     14:52                Code
d-----        2024/07/16     11:56                DBeaverData
d---s-        2024/10/15     14:37                Microsoft
d-----        2024/07/09     11:21                teraterm5
```

**Reply by schneebuzz** [reply]
This doesn't seem like a problem with Poetry. Could you confirm that you are operating on the same directory? On Windows, the installation is performed to `%AppData%` unless overridden through environment variables. Therefore, the installation depends on the executing user.

**Reply by gyuta** [reply]
Thank you for your feedback. I understand that this is not a problem related to Poetry. However, I chose this location because I'm not aware of a more appropriate place, and I hope that other users who have faced similar issues while installing Poetry on Windows will see this.

I have confirmed that I am operating in the same directory, but I’m still struggling to resolve the issue. If you have any other suggestions or advice, I would greatly appreciate it.
